_rules/SC3-1-2-lang-valid.md

```diff
@@ -0,0 +1,80 @@
+---
+name: Valid body lang attribute
+
+description:
+- This rule checks that `lang` or `xml:lang` attributes on elements within the `body` of a web page conforms to [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt).
+
+success_criterion:
+- 3.1.2
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Bryn Anderson
+- Jey Nandakumar
+---
+
+## Test Procedure
+
+### Applicability
+
+Any DOM element, within the `body` of a webpage with a [non-empty](#non-empty) `lang` or `xml:lang` attribute.
+
+**Note**: To retrieve all DOM elements with a `lang` or `xml:lang` attribute, other than the root element use -
+`descendent-or-self::body[@lang]` -or-  `descendent-or-self::body[@xml:lang]`.
+
+### Expectation
+
+The value of the `lang` or `xml:lang` conforms to [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt).
+
+## Assumptions
+
+*There are currently no assumptions*
+
+## Background
+
+- https://www.w3.org/TR/2014/NOTE-WCAG20-TECHS-20140408/H58
+- https://www.ietf.org/rfc/bcp/bcp47.txt
+- http://wiki.egovmon.no/wiki/SC3.1.2#Element_descendent-or-self::body.5B.40lang.5D_or_descendent-or-self::body.5B.40xml:lang.5D
+- https://www.w3.org/WAI/WCAG20/quickref/?showtechniques=312#qr-meaning-other-lang-id
+
+## Test Cases
+
+### Passed
```

Comment:
None of the tast cases here are placed in the body, so actually they are not applicable. 
I suggest placing all of the current test cases within a <body>, and then adding a single inapplicable test case where it is not placed within the body, but e.g. on the <html> element.